data-prepper-plugins/date-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/date/DateProcessorConfig.java

```diff
@@ -82,30 +93,57 @@ public static boolean isValidPattern(final String pattern) {
     }
 
     @JsonProperty("from_time_received")
+    @JsonPropertyDescription("When `true`, the timestamp from the event metadata, " +
+            "which is the time at which the source receives the event, is added to the event data. " +
+            "This option cannot be defined at the same time as `match`. Default is `false`.")
     private Boolean fromTimeReceived = DEFAULT_FROM_TIME_RECEIVED;
 
     @JsonProperty("to_origination_metadata")
+    @JsonPropertyDescription("When `true`, the matched time is also added to the event's metadata as an instance of " +
+            "`Instant`. Default is `false`.")
     private Boolean toOriginationMetadata = DEFAULT_TO_ORIGINATION_METADATA;
 
     @JsonProperty("match")
+    @JsonPropertyDescription("The date match configuration. " +
+            "This option cannot be defined at the same time as `from_time_received`. There is no default value.")
     private List<DateMatch> match;
 
     @JsonProperty("destination")
+    @JsonPropertyDescription("The field used to store the timestamp parsed by the date processor. " +
+            "Can be used with both `match` and `from_time_received`. Default is `@timestamp`.")
     private String destination = DEFAULT_DESTINATION;
 
     @JsonProperty("output_format")
+    @JsonPropertyDescription("Determines the format of the timestamp added to an event. " +
+            "Default is `yyyy-MM-dd'T'HH:mm:ss.SSSXXX`.")
     private String outputFormat = DEFAULT_OUTPUT_FORMAT;
 
     @JsonProperty("source_timezone")
+    @JsonPropertyDescription("The time zone used to parse dates, including when the zone or offset cannot be extracted " +
```

Comment:
When we update to Java 17 or 21 at a minimum (see #3192), we can start supporting multiline values. this should be nicer.

https://blogs.oracle.com/javamagazine/post/text-blocks-come-to-java

Reply:
It seems to me we have to rejoin the text even if we split into multilines?